src/Threads.cpp

```diff
@@ -134,6 +131,9 @@ void CThreadPool::threadFunc() {
 	assert(m_num_threads > 0 && m_num_idle > 0);
 	m_num_threads--;
 	m_num_idle--;
+
+	if (m_num_threads == 0)
```

Comment:
I mean that thread may go out from while loop by 2 reasons: m_done, and if the thread is idle.
If at some point ZNC starts killing idle threads after some timeout for example, the last killed by this reason thread will trigger the condition var.
Of course, m_num_threads needs to be checked, I didn't mean m_done only :)
`if (m_num_threads == 0 && m_done)` would help.